bpf/lib/egress_gateway.h

```diff
@@ -22,6 +24,23 @@
 #define EGRESS_GATEWAY_NO_GATEWAY (0)
 #define EGRESS_GATEWAY_EXCLUDED_CIDR bpf_htonl(1)
 
+static __always_inline
+int egress_gw_fib_lookup_and_redirect(struct __ctx_buff *ctx, __be32 egress_ip, __s8 *ext_err)
+{
+	struct bpf_fib_lookup_padded fib_params = {};
+	__u32 old_oif = ctx_get_ifindex(ctx);
+
+	*ext_err = (__s8)fib_lookup_v4(ctx, &fib_params, egress_ip, 0, 0);
```

Comment:
I thought it was ok to set just the egress/source IP as we want to identify the interface with that address assigned to (and that should be independent from the destination), or do you have a specific case in mind where this would not not work?

Reply:
Hmm not in combination with how we do EgressGW. We already say that all selected traffic should exit through *one* specific interface.

I'm just a bit wary about having *one* caller in the codebase that doesn't set the `daddr`. That makes it easy to miss bugs / corner cases, because no other path would be exposed to it.

Reply:
> I'm just a bit wary about having one caller in the codebase that doesn't set the daddr. That makes it easy to miss bugs / corner cases, because no other path would be exposed to it.

yeah good point, I'll just pass `tuple.daddr` as well :+1: